### PR TITLE
Add initial glossary of IDE terms

### LINKED ISSUE #13426: IDE terminology
I think it would be helpful if there was a page with the most common IDE terminology Roslyn is involved with or feeds to, as either I don't get it right or it is used inconsistently, perhaps as it evolved.

Examples of terminology I am thinking about:
- the light bulb and its entries (is it quick action? quick fix? refactorings? smart tag?) and previews
- the colored bars on the left margin of the code
- what is tooltip, quick info, intellisense vs. autocompletion, filters
- the three dropdowns on top of the editor
- squiggles that mark code
- suggestion / completion mode
- codelens and their anatomy
- collapsible blocks and the three dots
- the dimmed code (simplifications)

etc. That would be useful for reporting bugs as well as communicating new features.

### PR DESCRIPTION
**Pull request by Pilchie**
Fixes #13426.

### CODE REVIEW

**Comment by Pilchie** [comment]
Tagging @dotnet/roslyn-ide and @miloush.

File: docs/ide/glossary.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
...ka **autocomplete** or **statement completion** - The dropdown of possible items to insert into the editor at the current location.  Generally triggered automatically when typing or pressing `Ctrl+J` or `Ctrl+Space`.  Can be in one of two modes, either *Completion mode* or *Suggestion mode*.  In *Completion mode* (the default) where typing any character that isn't part of the currently selected entry "commits" the current entry and inserts it's text. *Suggestion mode* completion shows a "builder", and the current item is "soft selected" (outlined, instead of fully selected).  In suggestion mode typing a character that isn't in the list doesn't commit the item, in order to commit you have to do it explicitly with tab/enter/double click.  You can toggle between *completion mode* and *suggestion mode* by pressing `Ctrl+Alt+Space`. In cases where it's ambiguous whether you are inserting a new identifier or referring to an existing one, the IDE generally forces *suggestion mode* so that you don't accidentally commit an existing item when trying to declare a new one (for example, in a context where a delegate is expected, suggestion mode is force to allow typing a lambda parameter name).
```
Comment (svick) [inline docs/ide/glossary.md]: >  (aka **autocomplete** or **statement completion** -

Missing `)`.

> and inserts it's text

Typo: it's → its.

> suggestion mode is force to allow typing

Typo: force → forced?

Comment (Pilchie) [inline docs/ide/glossary.md]: Thanks @svick!  Fixed in latest commits.

**Comment by Pilchie** [comment]
@dotnet-bot retest  windows_debug_unit64_prtest please - logs gone.

**Pull request merged**